fre/pp/configure_script_yaml.py

```diff
@@ -1,124 +1,256 @@
-          rose_regrid.set(keys=[f'{comp}', 'inputRealm'], value=f'{i.get("inputRealm")}')
-          rose_regrid.set(keys=[f'{comp}', 'inputGrid'], value=f'{i.get("sourceGrid")}')
-          rose_regrid.set(keys=[f'{comp}', 'interpMethod'], value=f'{interp_method}')
-          interp_split = i.get('xyInterp').split(',')
-          rose_regrid.set(keys=[f'{comp}', 'outputGridLon'], value=f'{interp_split[1]}')
-          rose_regrid.set(keys=[f'{comp}', 'outputGridLat'], value=f'{interp_split[0]}')
-          rose_regrid.set(keys=[f'{comp}', 'outputGridType'], value=f'{interp_split[0]}_{interp_split[1]}.{interp_method}')
-
-  # write output files
-  print("Writing output files...")
-  cylc_dir = os.path.join(os.path.expanduser("~/cylc-src"), f"{e}__{p}__{t}")
-
-  outfile = os.path.join(cylc_dir, 'pp.yaml')
-  shutil.copyfile(yml, outfile)
-  print("  " + outfile)
-
-  dumper = metomi.rose.config.ConfigDumper()
-  outfile = os.path.join(cylc_dir, "rose-suite.conf")
-  dumper(rose_suite, outfile)
-  print("  " + outfile)
-
-  outfile = os.path.join(cylc_dir, "app", "regrid-xy", "rose-app.conf")
-  dumper(rose_regrid, outfile)
-  print("  " + outfile)
-
-  outfile = os.path.join(cylc_dir, "app", "remap-pp-components", "rose-app.conf")
-  dumper(rose_remap, outfile)
-  print("  " + outfile)
+        if i.get("xyInterp") is not None:
+            rose_regrid.set(keys=[f'{comp}', 'sources'], value=f'{sources}')
+            rose_regrid.set(keys=[f'{comp}', 'inputRealm'], value=f'{i.get("inputRealm")}')
+            rose_regrid.set(keys=[f'{comp}', 'inputGrid'], value=f'{i.get("sourceGrid")}')
+            rose_regrid.set(keys=[f'{comp}', 'interpMethod'], value=f'{interp_method}')
+            interp_split = i.get('xyInterp').split(',')
+            rose_regrid.set(keys=[f'{comp}', 'outputGridLon'], value=f'{interp_split[1]}')
+            rose_regrid.set(keys=[f'{comp}', 'outputGridLat'], value=f'{interp_split[0]}')
+            rose_regrid.set(keys=[f'{comp}', 'outputGridType'],
+                            value=f'{interp_split[0]}_{interp_split[1]}.{interp_method}')
+
+####################
+def _yamlInfo(yamlfile,experiment,platform,target):
+    """
+    Using a valid pp.yaml, the rose-app and rose-suite
+    configuration files are created in the cylc-src
+    directory. The pp.yaml is also copied to the
+    cylc-src directory.
+    """
+    e = experiment
+    p = platform
+    t = target
+    yml = yamlfile
+
+    # Initialize the rose configurations
+    rose_suite,rose_regrid,rose_remap = rose_init(e,p,t)
+
+    # Combine main and exp yamls into one new combine.yaml
+    comb_yaml = consolidate_yamls(mainyaml=yml,
+                                  experiment=e,
+                                  platform=p,
+                                  target=t)
+
+    # Load the combined yaml
+    final_yaml = yaml_load(comb_yaml)
+
+    # Clean combined yaml to validate
+    del final_yaml["fre_properties"]
+    del final_yaml["shared"]
```

Comment:
It's possible (but unlikely) that pp yamls could have no `fre_properties` or `shared` sections, do you agree?

If those two keys don't exist, then the `del` will cause an exception.

I think a `experiments` section is always required, though.

Reply:
Hmm I suppose there could be that case. I could add an check for that.